src/ALE/MOM_remapping.F90

```diff
@@ -1415,387 +1670,794 @@ logical function remapping_unit_tests(verbose)
+  call reintegrate_column(nsrc, h_src, uh_src, ndest, h_dest, uh_dest)
+  call test%real_arr(ndest, uh_dest, uh_true, msg)
+
+end subroutine test_reintegrate
+
+! =========================================================================================
+! The following provide the function for the testing_type helper class
+
+!> Update the state with "test"
+subroutine test(this, state, label)
+  class(testing),   intent(inout) :: this  !< This testing class
+  logical,          intent(in)    :: state !< True to indicate a fail, false otherwise
+  character(len=*), intent(in)    :: label !< Message
+
+  this%num_tests_checked = this%num_tests_checked + 1
+  if (state) then
+    this%state = .true.
+    this%num_tests_failed = this%num_tests_failed + 1
+    this%ifailed( this%num_tests_failed ) = this%num_tests_checked
+    if (this%num_tests_failed == 1) this%label_first_fail = label
+  endif
+  if (this%stop_instantly .and. this%state) stop 1
+end subroutine test
+
+!> Set attributes
+subroutine set(this, verbose, stdout, stderr, stop_instantly)
+  class(testing), intent(inout) :: this  !< This testing class
+  logical, optional, intent(in) :: verbose !< True or false setting to assign to verbosity
+  integer, optional, intent(in) :: stdout !< The stdout channel to use
+  integer, optional, intent(in) :: stderr !< The stderr channel to use
+  logical, optional, intent(in) :: stop_instantly !< If true, stop immediately on error detection
+
+  if (present(verbose)) then
+    this%verbose = verbose
+  endif
+  if (present(stdout)) then
+    this%stdout = stdout
+  endif
+  if (present(stderr)) then
+    this%stderr = stderr
+  endif
+  if (present(stop_instantly)) then
+    this%stop_instantly = stop_instantly
+  endif
+end subroutine set
+
+!> Returns state
+logical function outcome(this)
+  class(testing), intent(inout) :: this !< This testing class
+  outcome = this%state
+end function outcome
+
+!> Summarize results
+logical function summarize(this, label)
+  class(testing),  intent(inout) :: this  !< This testing class
+  character(len=*),   intent(in) :: label !< Message
+  integer :: i
+
+  if (this%state) then
+    write(this%stdout,'(a," : ",a,", ",i4," failed of ",i4," tested")') &
+         red('FAIL'), trim(label), this%num_tests_failed, this%num_tests_checked
+    write(this%stdout,'(a,100i4)') 'Failed tests:',(this%ifailed(i),i=1,this%num_tests_failed)
+    write(this%stdout,'(a,a)') 'First failed test: ',trim(this%label_first_fail)
+    write(this%stderr,'(a,100i4)') 'Failed tests:',(this%ifailed(i),i=1,this%num_tests_failed)
+    write(this%stderr,'(a,a)') 'First failed test: ',trim(this%label_first_fail)
+    write(this%stderr,'(a," : ",a)') trim(label),'FAILED'
+  else
+    write(this%stdout,'(a," : ",a,", all ",i4," tests passed")') &
+         green('Pass'), trim(label), this%num_tests_checked
+  endif
+  summarize = this%state
+end function summarize
+
+!> Pads string with the color codes for red/reset
```

Comment:
Should the functions `red()` and `green()` be placed in the MOM_string_functions module at src/framework
/MOM_string_functions.F90 instead of (or in addition to) the MOM_remapping module, to facilitate their remotes more widely with in the MOM6 code, perhaps with the the more explicit names `red_string()` and `green_string()`?